app/react/portainer/settings/SettingsView/KubeSettingsPanel/validation.ts

```diff
@@ -0,0 +1,28 @@
+import { SchemaOf, object, string, boolean, number } from 'yup';
+
+import { isValidUrl } from '@@/form-components/validate-url';
+
+import { FormValues } from './types';
+
+export function validation(): SchemaOf<FormValues> {
+  return object().shape({
+    helmRepositoryUrl: string()
+      .default('')
+      .test('valid-url', 'Invalid URL', (value) => !value || isValidUrl(value)),
+    kubeconfigExpiry: string().required(),
+    globalDeploymentOptions: object().shape({
+      hideAddWithForm: boolean().required(),
+      perEnvOverride: boolean().required(),
+      hideWebEditor: boolean().required(),
+      hideFileUpload: boolean().required(),
+      requireNoteOnApplications: boolean().required(),
+      minApplicationNoteLength: number()
+        .typeError('Must be a number')
+        .default(0)
+        .when('requireNoteOnApplications', {
+          is: true,
+          then: (schema) => schema.required().min(1, 'Minimum value is 1'),
```

Comment:
While testing, I noticed that this field also has a maximum value in the validation message of the old version as shown below

![image](https://github.com/portainer/portainer/assets/100548325/7bef1ab8-2b4f-4d53-8638-296ce034e148)